sdg/ProgressMeasure.py

```diff
@@ -61,8 +65,10 @@ def get_indicator_progress(self):
                     progress_outputs.append((score, pm.status))
 
             if progress_outputs:
-                # Return a tuple of the minimum score and associated progress status
-                return min(progress_outputs, key=lambda x: x[0])
+                # Cache/return a tuple of the minimum score and associated progress status
+                results = min(progress_outputs, key=lambda x: x[0])
+                self.cache_store[self.inid] = results
```

Comment:
Leads to a TypeError if self.cache_store is None. Replace line 70 with:
```
if self.cache_store is None:
  self.cache_store = {self.inid: results}
else:
  self.cache_store[self.inid] = results
```

Reply:
Actually, I will just merge and edit the lines on my end. It will probably be faster this way.